### PR TITLE
[examples] add cmake for gp712, qpg6100, qpg7015m

### PR DESCRIPTION
**Pull request by nikita-s-wrk**
Added Qorvo platforms to cmake.
Updated git actions
Updated bootstrap to include arm-linux-gnueabihf toolchain

### CODE REVIEW

File: examples/platforms/gp712/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
```
Comment (jwhui) [inline examples/platforms/gp712/CMakeLists.txt:2]: Here and other new files below:
```suggestion
#  Copyright (c) 2021, The OpenThread Authors.
```

File: examples/platforms/gp712/arm-linux-gnueabihf.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set(CMAKE_SYSTEM_NAME              Generic)
+set(CMAKE_SYSTEM_PROCESSOR         ARM)
+
+set(CMAKE_C_COMPILER               arm-linux-gnueabihf-gcc)
```
Comment (jwhui) [inline examples/platforms/gp712/arm-linux-gnueabihf.cmake:32]: FYI - this name seems to have changed to `arm-none-linux-gnueabihf-gcc` in v9 and later.

File: script/bootstrap [hunk 3]
```diff
@@ -65,7 +65,7 @@ install_packages_apt()
     if [ "$PLATFORM" = "Raspbian" ]; then
         sudo apt-get --no-install-recommends install -y binutils-arm-none-eabi gcc-arm-none-eabi gdb-arm-none-eabi libnewlib-arm-none-eabi libstdc++-arm-none-eabi-newlib
     elif [ "$PLATFORM" = "Ubuntu" ]; then
-        sudo apt-get --no-install-recommends install -y ca-certificates wget
+        sudo apt-get --no-install-recommends install -y ca-certificates wget gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf
```
Comment (jwhui) [inline script/bootstrap:68]: Do you mind avoiding this for now and adding this into the "Toolchain Setup" section of the `README.md` for the platforms that require it?

**Review by jwhui — COMMENTED** [review]
Looks great overall, thanks! I posted some minor comments.

File: examples/platforms/gp712/arm-linux-gnueabihf.cmake (hunk 2 above)
Comment (nikita-s-wrk) [reply examples/platforms/gp712/arm-linux-gnueabihf.cmake:32]: Thanks for note,
We are currently targeting arm-linux-gnueabihf-gcc toolchain with qpg7015m and gp712 platforms. 
However, we will discuss how we could align toolchain versions and propose an update for cmake and CI scripts if needed in the future.

**Comment by nikita-s-wrk** [comment]
`codecov/project — 77.90% (-9.18%)`
Fails randomly - no changes were made in referenced files.

`Simulation 1.2 / thread-border-router (pull_request) Failing after 12m — thread-border-router`
Simulation test failed via "BrokenPipeError: [Errno 32] Broken pipe" (Issue: https://github.com/openthread/openthread/issues/6264)

**Review by jwhui — APPROVED** [review]

**Pull request merged**